### DISCUSSION TITLE
更新模板后测试报错sysy/sylib.h: No such file or directory

### DISCUSSION THREAD

**Discussion by SundayDream00** [arcsysu/SYsU-lang#23] [Q&A] [answered]
如题。对仓库进行了完整克隆，将lexer.l替换为之前通过测试的文件，lib/sysy文件夹下存在sylib.h，但测试报错找不到文件，请问如何解决

**Comment by wu-kan** [comment, marked as answer]
测试的时候有设置环境变量 CPATH 吗

**Reply by SundayDream00** [reply]
原来如此，之前只按测试报错的旧命令执行，加了环境变量之后就解决了，感谢助教

**Reply by SundayDream00** [reply]
![1](https://user-images.githubusercontent.com/98109054/159660324-53df652e-7416-492d-b734-ae46538967ae.png)
请问是每次单独执行某文件的评测时都需要进行路径设置吗，如果重复模板中的PATH、CPATH和LD_LIBRARY_PATH设置命令会得到如图所示报错

**Reply by wu-kan** [reply]
第三行 `~/sysu/ lib` 这里似乎多了一个空格。

**Reply by wu-kan** [reply]
另外测试指令外面最好包一层括号，让设置的环境变量只对当前指令有效，防止污染你的环境

**Reply by SundayDream00** [reply]
如果上述情况都修改之后仍报相同错误的话是环境已被污染吗

**Reply by wu-kan** [reply]
检查下输入指令每行行末是否有多余空格，`\` 之后不能有空格

**Reply by Aurora-1412** [reply]
请问在哪里设置环境变量

**Reply by wu-kan** [reply]
> 请问在哪里设置环境变量

README.md 里面每条测试语句前的 export 就是在设置环境变量
